### PR TITLE
[feat] FMC: extend PCRs

### PR DESCRIPTION
**Pull request by rusty1968**
- FMC reads the measurement of the Runtime FW Module, TCIRT, from the Data Vault that has previously been validated by ROM.
- FMC reads SVN RT from the Data Vault that has previously been validated by ROM.
- FMC extends Caliptra PCR registers with TCIRT.
- FMC extends Caliptra PCR registers with SVN RT.

### CODE REVIEW

File: fmc/src/flow/rt_alias.rs [hunk 1]
```diff
@@ -11,23 +11,37 @@ Abstract:
...
 pub struct RtAliasLayer {}
 
 impl DiceLayer for RtAliasLayer {
     /// Perform derivations for the DICE layer
-    fn derive(env: &FmcEnv, _: &DiceInput) -> CaliptraResult<DiceOutput> {
-        Self::extend_pcrs(env)?;
-        Err(0)
+    fn derive(env: &FmcEnv, _input: &DiceInput) -> CaliptraResult<DiceOutput> {
+        // At this point PCR0 & PCR1 must have the same value. We use the value
+        // of PCR1 as the UDS for deriving the CDI
```
Comment (bluegate010) [inline fmc/src/flow/rt_alias.rs:28]: PCRs hold measurements, shouldn't we be looking in KV slots for UDS / CDI? Also FMC doesn't have access to UDS.

Comment (rusty1968) [reply fmc/src/flow/rt_alias.rs:28]: This is a comment from the code I lifted from ROM, which is quite confusing. PCR1 holds measurements but they call it UDS but it should  be the CDI of the previous layer.

File: fmc/src/hand_off.rs [hunk 2]
```diff
@@ -86,6 +89,27 @@ impl HandOff {
         unsafe { transfer_control(rt_entry) }
     }
 
+    /// Retrieve runtime TCI (digest)
+    pub fn rt_tci(&self, env: &FmcEnv) -> Array4x12 {
```
Comment (bluegate010) [inline fmc/src/hand_off.rs:93]: Data Vault has methods for these; could be `env.data_vault().map(|d| d.rt_tci())`, `env.data_vault().map(|d| d.rt_svn())`

Comment (rusty1968) [reply fmc/src/hand_off.rs:93]: We need to take the handoff table information into account as it decouples ROM and FMC. We can't hardcode their locations in enums as the data vault driver does.

Comment (bluegate010) [reply fmc/src/hand_off.rs:93]: Ah, gotcha.

File: fmc/src/flow/rt_alias.rs [hunk 3]
```diff
@@ -11,33 +11,63 @@ Abstract:
...
+        // At this point PCR0 & PCR1 must have the same value. We use the value
+        // of PCR1 as the data for deriving the CDI
+        let _measurement = env
+            .pcr_bank()
+            .map(|p| p.read_pcr(caliptra_drivers::PcrId::PcrId1));
+
+        // TODO : implement derivation.
+        // Derive the Rt layer CDI.
+        //let cdi = Self::derive_cdi(env, _measurement, input.cdi)?;
+        Err(0xdead)
```
Comment (jhand2) [inline fmc/src/flow/rt_alias.rs:36]: If this gets merged will FMC fail to boot? Or is this code not linked yet?

Comment (rusty1968) [reply fmc/src/flow/rt_alias.rs:36]: It is not not being invoked.  Our integration test that jumps from ROM to FMC to RT is successfully running on CI as expected.

Comment (rusty1968) [reply fmc/src/flow/rt_alias.rs:36]: @jhand2 , does the current/journey PCR allocation match your expectation ?

**Review by jhand2 — DISMISSED** [review]

File: fmc/src/flow/rt_alias.rs [hunk 4]
```diff
@@ -11,33 +11,63 @@ Abstract:
...
 pub struct RtAliasLayer {}
 
 impl DiceLayer for RtAliasLayer {
     /// Perform derivations for the DICE layer
-    fn derive(env: &FmcEnv, _: &DiceInput) -> CaliptraResult<DiceOutput> {
-        Self::extend_pcrs(env)?;
-        Err(0)
+    fn derive(env: &FmcEnv, _input: &DiceInput) -> CaliptraResult<DiceOutput> {
+        // At this point PCR0 & PCR1 must have the same value. We use the value
+        // of PCR1 as the data for deriving the CDI
```
Comment (FerralCoder) [inline fmc/src/flow/rt_alias.rs:28]: out of date comment

File: fmc/src/flow/pcr.rs [hunk 5]
```diff
@@ -30,17 +30,17 @@ use caliptra_drivers::{CaliptraResult, PcrId};
 /// # Arguments
 ///
 /// * `env` - FMC Environment
-pub fn extend_pcr0(env: &FmcEnv, hand_off: &HandOff) -> CaliptraResult<()> {
-    extend_pcr_common(env, hand_off, PcrId::PcrId0)
+pub fn extend_current_pcr(env: &FmcEnv, hand_off: &HandOff) -> CaliptraResult<()> {
+    extend_pcr_common(env, hand_off, PcrId::PcrId3)
 }
 
 /// Extend PCR1
```
[the remark this answers is not in the document]
Comment (rusty1968) [reply fmc/src/flow/pcr.rs]: updated.

File: fmc/src/flow/pcr.rs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    pcr.rs
+
+Abstract:
+
+    File contains execution routines for extending PCR0 & PCR1
```
[the remark this answers is not in the document]
Comment (rusty1968) [reply fmc/src/flow/pcr.rs]: updated.

File: fmc/src/flow/pcr.rs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+Environment:
+
+    ROM
+
+Note:
+
+    PCR0 - Journey PCR unlocked and cleared on cold reset
+    PCR1 - Current PCR unlocked and cleared on any reset
```
[the remark this answers is not in the document]
Comment (rusty1968) [reply fmc/src/flow/pcr.rs]: updated.

File: fmc/src/flow/rt_alias.rs (hunk 4 above)
Comment (rusty1968) [reply fmc/src/flow/rt_alias.rs]: updated.

**Review by FerralCoder — APPROVED** [review]

**Pull request merged**